flow/activities/flowable.go

```diff
@@ -544,36 +544,26 @@ func (a *FlowableActivity) CleanupQRepFlow(ctx context.Context, config *protos.Q
 	return dst.CleanupQRepFlow(ctx, config)
 }
 
-func (a *FlowableActivity) DropFlowSource(ctx context.Context, config *protos.ShutdownRequest) error {
-	sourcePeerName, err := a.getPeerNameForMirror(ctx, config.FlowJobName, Source)
-	if err != nil {
-		return err
-	}
-
-	ctx = context.WithValue(ctx, shared.FlowNameKey, config.FlowJobName)
+func (a *FlowableActivity) DropFlowSource(ctx context.Context, flowJobName string, sourcePeerName string) error {
```

Comment:
lets use protos as inputs and outputs for activities. Better flexibility to change things.